bundles/framework/myplacesimport/service/MyPlacesImportService.js

```diff
@@ -33,74 +31,223 @@ Oskari.clazz.define('Oskari.mapframework.bundle.myplacesimport.MyPlacesImportSer
+            errorCb();
+        });
+    },
+
+    updateUserLayer: function (layerId, values, successCb, errorCb) {
+        const id = this.getActualId(layerId);
+        fetch(this.getEditLayerUrl(id, values), {
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json'
+            }
+        }).then(response => {
+            if (!response.ok) {
+                throw Error(response.statusText);
+            }
+            return response.json();
+        }).then(json => {
+            this.updateLayer(json);
+            this._showSuccess('tab.notification.editedMsg');
+            successCb();
+        }).catch(error => {
+            this.log.error(error);
+            this._showError('tab.error.editMsg');
+            errorCb();
+        });
+    },
+    _handleErrorResponse: function (info, errorCb) {
+        const { error, extensions = [], cause, parser } = info || {};
+        let errorKey = error || ERRORS.GENERIC;
+        // Parser error has cause which is used for localized message
+        if (error === ERRORS.PARSER) {
+            if (cause === ERRORS.NO_SRS) {
+                errorKey = parser === 'shp' ? 'shpNoSrs' : 'noSrs';
+            } else if (cause === ERRORS.FORMAT) {
+                errorKey = cause;
+            }
+        }
+        // pass args for localization even them aren't needed for requested errorKey
+        const args = {
+            maxSize: this.instance.getMaxSize(),
+            extensions: extensions.join(',')
+        };
+        this._showError(`flyout.error.${errorKey}`, args);
+        // Only unknown srs is handled differently, use cause for callback
+        errorCb(cause);
+    },
+    _showError: function (locKey, args) {
+        const content = this.instance.loc(locKey, args);
+        Messaging.error({ content, duration: 10 });
     },
 
+    _showSuccess: function (locKey, args) {
+        const content = this.instance.loc(locKey, args);
+        Messaging.success({ content, duration: 10 });
+    },
+    _showWarning: function (warning = {}) {
+        const { featuresSkipped } = warning;
+        if (!featuresSkipped) {
+            return;
+        }
+        const content = this.instance.loc('flyout.warning.features_skipped', { count: featuresSkipped });
+        Messaging.warn({ content, duration: 10 });
+    },
     /**
-     * Retrieves the user layers (with the id param only the specified layer)
+     * Retrieves the user layers
      * from the backend and adds them to the map layer service.
      *
      * @method getUserLayers
-     * @param  {Function} successCb (optional)
-     * @param  {Function} errorCb (optional)
-     * @param  {String} id (optional)
      */
-    getUserLayers: function (successCb, errorCb, id) {
-        const me = this;
-        let url = this.urls.get;
-
-        if (id) {
-            url = url + '&id=' + id;
-        }
-
+    getUserLayers: function (id) {
         jQuery.ajax({
```

Comment:
We should update this and delete to fetch as well so we don't use both jQuery.ajax() and fetch()

Reply:
updated to fetch